### PR TITLE
Use spa-templates repo as a submodule

### PR DESCRIPTION
**Pull request by dougbu**
- see #32027

### CODE REVIEW

File: src/ProjectTemplates/test/ProjectTemplates.Tests.csproj [hunk 1]
```diff
@@ -56,7 +56,7 @@
     <ProjectReference Include="../Web.Client.ItemTemplates/Microsoft.DotNet.Web.Client.ItemTemplates.csproj" ReferenceOutputAssembly="false" />
     <ProjectReference Include="../Web.ItemTemplates/Microsoft.DotNet.Web.ItemTemplates.csproj" ReferenceOutputAssembly="false" />
     <ProjectReference Include="../Web.ProjectTemplates/Microsoft.DotNet.Web.ProjectTemplates.csproj" ReferenceOutputAssembly="false" />
-    <ProjectReference Include="../Web.Spa.ProjectTemplates/Microsoft.DotNet.Web.Spa.ProjectTemplates.csproj" ReferenceOutputAssembly="false" />
+    <ProjectReference Include="$(RepoRoot)src\submodules\spa-templates\src\Microsoft.DotNet.Web.Spa.ProjectTemplates.csproj" ReferenceOutputAssembly="false" />
```
Comment (dougbu) [inline src/ProjectTemplates/test/ProjectTemplates.Tests.csproj:59]: This reference to Microsoft.DotNet.Web.Spa.ProjectTemplates.csproj is insufficient to restore, build, or pack that template project when using `src\ProjectTemplates\build.cmd` or `src/ProjectTemplates/build.sh`. I had to modify those files ☹️

File: src/submodules/Directory.Build.props [hunk 2]
```diff
@@ -1,2 +1,9 @@
-<!-- Intentionally breaks the inheritance chain so submodules do not inherit build settings from this repo. -->
-<Project />
+<Project>
+  <!--
+    For everything but Microsoft.DotNet.Web.Spa.ProjectTemplates, intentionally break the inheritance chain so
+    submodules do not inherit build settings from this repo. For Microsoft.DotNet.Web.Spa.ProjectTemplates, break the
+    build if referenced file does not exist.
+  -->
+  <Import Project="..\ProjectTemplates\Directory.Build.props"
+      Condition=" '$(MSBuildProjectName)' == 'Microsoft.DotNet.Web.Spa.ProjectTemplates' " />
```
Comment (dougbu) [inline src/submodules/Directory.Build.props:8]: When I update the spa-template repo to build outside the aspnetcore repo, I'll make this logic unnecessary.

File: src/ProjectTemplates/build.cmd [hunk 3]
```diff
@@ -1,3 +1,4 @@
 @ECHO OFF
 SET RepoRoot=%~dp0..\..
 %RepoRoot%\eng\build.cmd -projects %~dp0*\*.*proj "/p:EnforceE2ETestPrerequisites=true" %*
+%RepoRoot%\eng\build.cmd -projects %~dp0\..\submodules\spa-templates\src\*.*proj "/p:EnforceE2ETestPrerequisites=true" %*
```
Comment (dougbu) [inline src/ProjectTemplates/build.cmd:4]:
```suggestion
%RepoRoot%\eng\build.cmd -projects %~dp0\..\submodules\spa-templates\src\*.*proj "/p:EnforceE2ETestPrerequisites=true" -noBuildRepoTasks -noBuildNative %*
```

File: src/ProjectTemplates/build.sh [hunk 4]
```diff
@@ -5,3 +5,4 @@ set -euo pipefail
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 repo_root="$DIR/../.."
 "$repo_root/eng/build.sh" --projects "$DIR/**/*.*proj" "/p:EnforceE2ETestPrerequisites=true" "$@"
+"$repo_root/eng/build.sh" --projects "$DIR/../submodules/spa-templates/src/*.*proj" "/p:EnforceE2ETestPrerequisites=true" "$@"
```
Comment (dougbu) [inline src/ProjectTemplates/build.sh:8]:
```suggestion
"$repo_root/eng/build.sh" --projects "$DIR/../submodules/spa-templates/src/*.*proj" --no-build-repo-tasks "/p:EnforceE2ETestPrerequisites=true" "$@"
```

**Review by wtgodbe — APPROVED** [review]
Seems good to me

**Pull request merged**